src/commands/configure/secrets.ts

```diff
@@ -0,0 +1,215 @@
+// has secrets for dependabot
+// has secrets for npm publish
+// if sign, has aws secrets
+// if nuts, has nuts auth secrets
+
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { SfCommand, Flags } from '@salesforce/sf-plugins-core';
+import { Messages } from '@salesforce/core';
+
+import { Octokit } from 'octokit';
+import { exec } from 'shelljs';
+import * as yaml from 'js-yaml';
+import * as fs from 'fs';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.load('@salesforce/plugin-dev', 'configure.secrets', [
+  'summary',
+  'description',
+  'examples',
+  'flags.repository.summary',
+  'flags.dryRun.summary',
+]);
+
+type SecretClassification =
+  | 'not needed'
+  | 'overridden by repo'
+  | 'shared to repo by org'
+  | 'exists, but not shared with repo'
+  | 'does not exist in org'
+  | 'unable to check';
+
+type GhaJob = {
+  uses: string;
+  with: {
+    ctc: boolean;
+    sign: boolean;
+  };
+};
+export type SecretsResult = {
+  AWS_ACCESS_KEY_ID: SecretClassification;
+  AWS_SECRET_ACCESS_KEY: SecretClassification;
+  NPM_TOKEN: SecretClassification;
+  TESTKIT_AUTH_URL: SecretClassification;
+  SF_CLI_BOT_GITHUB_TOKEN: SecretClassification;
+  SF_CHANGE_CASE_SFDX_AUTH_URL: SecretClassification;
+};
+
+export default class ConfigureSecrets extends SfCommand<SecretsResult> {
+  public static summary = messages.getMessage('summary');
+  public static description = messages.getMessage('description');
+  public static examples = messages.getMessages('examples');
+
+  public static flags = {
+    repository: Flags.string({
+      summary: messages.getMessage('flags.repository.summary'),
+      char: 'r',
+      required: true,
+    }),
+    'dry-run': Flags.boolean({
+      summary: messages.getMessage('flags.dryRun.summary'),
+      char: 'c',
+    }),
+  };
+
+  public async run(): Promise<SecretsResult> {
+    const { flags } = await this.parse(ConfigureSecrets);
+    const output: SecretsResult = {
+      AWS_ACCESS_KEY_ID: 'not needed',
+      AWS_SECRET_ACCESS_KEY: 'not needed',
+      NPM_TOKEN: 'not needed',
+      TESTKIT_AUTH_URL: 'not needed',
+      SF_CLI_BOT_GITHUB_TOKEN: 'not needed',
+      SF_CHANGE_CASE_SFDX_AUTH_URL: 'not needed',
+    };
+
+    // TODO: nice error if no token exists
+    const octokit = new Octokit({ auth: process.env.GITHUB_TOKEN });
+
+    const repoBase = {
+      owner: flags.repository.split('/')[0],
+      repo: flags.repository.split('/')[1],
+    };
+
+    // clone repo so we can review its yaml
+    exec(`git clone https://github.com/${flags.repository}`);
+
+    // part 1: what secrets does this repo need?
+    const publish = yaml.load(
+      await fs.promises.readFile(`${repoBase.repo}/.github/workflows/onRelease.yml`, 'utf8')
```

Comment:
Unless you need the git history you can do a `git clone <repo> --depth 1`:
https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/How-and-when-to-perform-a-depth-1-git-clone

Even better, if you only need the file contents of a repo you can use the GH api instead of cloning:
https://docs.github.com/en/rest/repos/contents#get-repository-content

```
gh api \
  -H 'Accept: application/vnd.github.VERSION.raw' \
  repos/salesforcecli/testPackageRelease/contents/.github/workflows/onRelease.yml

** raw yaml **
```

Reply:
I'm gonna go with the depth 1 so we can get the files all in one go